icu4j/main/classes/core/src/com/ibm/icu/text/PersonNameFormatter.java

```diff
@@ -0,0 +1,370 @@
+
+        /**
+         * Returns the appropriate NameField for its display name.
+         * @internal
+         */
+        public static NameField forString(String name) {
+            for (NameField field : values()) {
+                if (field.name.equals(name)) {
+                    return field;
+                }
+            }
+            throw new IllegalArgumentException("Invalid field name " + name);
+        }
+    }
+
+    /**
+     * Identifiers for the name field modifiers supported by the PersonName and PersonNameFormatter objects.
+     * @internal
+     */
+    public enum FieldModifier {
+        /**
+         * Requests an "informal" variant of the field, generally a nickname of some type:
+         * if "given" is "James", "given-informal" might be "Jimmy".  Only applied to the "given"
+         * field.  If the PersonName object doesn't apply this modifier, PersonNameFormatter just
+         * uses the unmodified version of "given".
+         * @internal
+         */
+        INFORMAL("informal"),
+
+        /**
+         * If the field contains a main word with one or more separate prefixes, such as
+         * "van den Hul", this requests just the prefixes ("van den").  Only applied to the "surname"
+         * field.  If the PersonName object doesn't apply this modifier, PersonNameFormatter
+         * assumes there are no prefixes.
+         * @internal
+         */
+        PREFIX("prefix"),
+
+        /**
+         * If the field contains a main word with one or more separate prefixes, such as
+         * "van den Hul", this requests just the main word ("Hul").  Only applied to the "surname"
+         * field.  If the implementing class doesn't apply this modifier, PersonNameFormatter
+         * assumes the entire "surname" field is the "core".
+         * @internal
+         */
+        CORE("core"),
+
+        /**
+         * Requests an initial for the specified field.  PersonNameFormatter will do
+         * this algorithmically, but a PersonName object can apply this modifier itself if it wants
+         * different initial-generation logic (or stores the initial separately).
+         * @internal
+         */
+        INITIAL("initial"),
+
+        /**
+         * Requests an initial for the specified field, suitable for use in a monogram
+         * (this usually differs from "initial" in that "initial" adds a period and "monogram" doesn't).
+         * PersonNameFormatter will do this algorithmically, but a PersonName object can apply
+         * this modifier itself if it wants different monogram-generation logic.
+         * @internal
+         */
+        MONOGRAM("monogram"),
+
+        /**
+         * Requests the field value converted to ALL CAPS.  PersonName objects
+         * generally won't need to handle this modifier themselves.
+         * @internal
+         */
+        ALL_CAPS("allCaps"),
+
+        /**
+         * Requests the field value with the first letter of each word capitalized.
```

Comment:
"... first grapheme of each word capitalized."

Reply:
Done.